src/Symfony/Component/DependencyInjection/ParameterBag/EnvPlaceholderParameterBag.php

```diff
@@ -71,6 +71,12 @@ public function getEnvPlaceholders()
      */
     public function mergeEnvPlaceholders(self $bag)
     {
-        $this->envPlaceholders = array_merge_recursive($this->envPlaceholders, $bag->getEnvPlaceholders());
+        $newPlaceholders = $bag->getEnvPlaceholders();
```

Comment:
B) is the only right answer. We though array_merge_recursive would do it, but you proved us wrong. Now we're trying to achieve B), nothing else :)

> just want to be sure I get this right this time.

I ran my proposal with your tests, so we're safe, isn't it ;)